test/common/testlib.py

```diff
@@ -1131,33 +1131,56 @@ class _DebugOutcome(unittest.case._Outcome):
     """
 
     def testPartExecutor(self, test_case, isTest=False):
-        # we want to do the debug actions after the test function got called (isTest==True), but before
-        # TestCase.run() calls tearDown(); thus remember whether we are after isTest and already ran
-        if isTest:
-            self.ran_debug = False
-        else:
-            if getattr(self, "ran_debug", None) is False:
-                self.ran_debug = True
-                if self.errors:
-                    (err_case, exc_info) = self.errors[-1]
-                    if exc_info and isinstance(test_case, MachineCase):
-                        assert err_case == test_case
-                        # strip off the two topmost frames for testPartExecutor and TestCase.run(); uninteresting and breaks naughties
-                        traceback.print_exception(exc_info[0], exc_info[1], exc_info[2].tb_next.tb_next)
-                        try:
-                            test_case.snapshot("FAIL")
-                            test_case.copy_js_log("FAIL")
-                            test_case.copy_journal("FAIL")
-                            test_case.copy_cores("FAIL")
-                        except (OSError, RuntimeError):
-                            # failures in these debug artifacts should not skip cleanup actions
-                            sys.stderr.write("Failed to generate debug artifact:\n")
-                            traceback.print_exc(file=sys.stderr)
-
-                        if opts.sit:
-                            sit(test_case.machines)
-
-        return super().testPartExecutor(test_case, isTest)
+        def failureHandler(exc_info, python_311=False):
+            if python_311:
+                # exc_info is now a string
```

Comment:
.. this is worth a Python bug report. It makes absolutely no sense for this to be a string, instead of the actual `exc_info` object..

Reply:
Maybe, but it is kinda an internal api.